### PR TITLE
Support custom switch-case indentation

### LINKED ISSUE #180239: Support TS custom switch-case indentation
Support issue for #179670

See https://github.com/microsoft/TypeScript/pull/53733 for details

### PR DESCRIPTION
**Pull request by donaldnevermore**
This PR correlates closely with microsoft/TypeScript#53733 due to API change. ~~Don't merge before microsoft/TypeScript#53733 gets merged. It won't work until the typescript version in vscode is upgraded.~~

Closes #180239, microsoft/TypeScript#18682

### CODE REVIEW

File: extensions/typescript-language-features/package.nls.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
 	"format.semicolons.ignore": "Don't insert or remove any semicolons.",
 	"format.semicolons.insert": "Insert semicolons at statement ends.",
 	"format.semicolons.remove": "Remove unnecessary semicolons.",
+	"format.indentSwitchCase": "Indent case clauses in switch statements.",
```
Comment (andrewbranch) [inline extensions/typescript-language-features/package.nls.json:42]: @mjbvz should individual formatter preferences indicate what TS version they require? I’m realizing now I didn’t do that on `format.semicolons` back when I added that.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts [hunk 2]
```diff
@@ -161,6 +161,7 @@ export default class FileConfigurationManager extends Disposable {
 			placeOpenBraceOnNewLineForFunctions: config.get<boolean>('placeOpenBraceOnNewLineForFunctions'),
 			placeOpenBraceOnNewLineForControlBlocks: config.get<boolean>('placeOpenBraceOnNewLineForControlBlocks'),
 			semicolons: config.get<Proto.SemicolonPreference>('semicolons'),
+			indentSwitchCase: config.get<boolean>('indentSwitchCase'),
```
[the remark this answers is not in the document]
Comment (donaldnevermore) [reply extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts:165]: OK.

**Review by andrewbranch — DISMISSED** [review]

File: extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (donaldnevermore) [reply extensions/typescript-language-features/src/languageFeatures/fileConfigurationManager.ts:164]: Now it's not needed any more.

**Review by andrewbranch — APPROVED** [review]

File: extensions/typescript-language-features/package.nls.json (hunk 1 above)
Comment (mjbvz) [reply extensions/typescript-language-features/package.nls.json:42]: Yes we should call this out. See some of the other setting strings for an example of this

**Review by mjbvz — APPROVED** [review]

**Review by rebornix — APPROVED** [review]

**Pull request merged**